modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -209,6 +259,23 @@ static GCompileArg exec(cv::gapi::s11n::IIStream& is, const std::string& tag) {
 }
 };
 
+template<typename T> struct deserialize_runarg;
+
+template<typename RMatAdapterType>
+struct deserialize_runarg {
+static GRunArg exec(cv::gapi::s11n::IIStream& is, uint32_t idx) {
+    if (idx == GRunArg::index_of<RMat>()) {
+        RMatAdapterType adapter = cv::gapi::s11n::detail::S11N<RMatAdapterType>::deserialize(is);
```

Comment:
Should adapter be supporting `operator=`/`copy constructor` in that case?

Reply:
Not sure if `S11N` is required here as the `RMatAdapterType` should implement `deserialize`

Reply:
Changed